doc/hardware/porting/board_porting.rst

```diff
@@ -8,17 +8,71 @@ directory* with various files in it. Files in the board directory inherit
 support for at least one SoC and all of its features. Therefore, Zephyr must
 support your :term:`SoC` as well.
 
+.. _board_and_identifiers:
+
+Board and board identifiers
+***************************
+
+A board may be a physical piece of hardware or an emulated board.
+Furthermore a board may contain one or multiple SoCs. Also, each SoC may contain
+one or multiple cpuclusters. Note, a cpucluster may refer to a single CPU core.
```

Comment:
you need to explain what a cpucluster with more details. You are assuming that cpucluster is obvious and established terminology, it is not.

Reply:
@tejlmand I second @nashif  comment. We need to understand if a cluster is a group of similar cores or different ones.

e.g a board with 4 arm cores and 1 DSP would have 2 clusters (4 x arm + 1 x dsp) or 1 cluster with all cores.

Reply:
Good points indeed.

Have improved the description and also mention the current design limitation in Zephyr.
Extending the model is something we can consider in the future.